Framework/Algorithms/src/FilterEvents.cpp

```diff
@@ -316,17 +320,15 @@ void FilterEvents::exec() {
 
   // Form the names of output workspaces
   std::vector<std::string> outputwsnames;
-  std::map<int, DataObjects::EventWorkspace_sptr>::iterator miter;
   Goniometer inputGonio = m_eventWS->run().getGoniometer();
-  for (miter = m_outputWorkspacesMap.begin();
-       miter != m_outputWorkspacesMap.end(); ++miter) {
+  for (auto miter : m_outputWorkspacesMap) {
```

Comment:
```suggestion
  for (auto &miter : m_outputWorkspacesMap) {
```

Reply:
Ah yes, good spot. Probably should be a const as well.